### PR TITLE
samples: mbedtls_sslclient: Use entropy driver to kickstart RNG

### PR DESCRIPTION
**Pull request by lpereira**
If an entropy driver is available during build, use it to gather
entropy rather than relying on sys_rand32_get(), which provides no
guarantees that the provided numbers will be good enough from an
encryption standpoint.

Signed-off-by: Leandro Pereira <leandro.pereira@intel.com>

### CODE REVIEW

**Review by jukkar — APPROVED** [review]
LGTM

File: samples/net/mbedtls_sslclient/src/mini_client.c [hunk 1]
```diff
@@ -139,28 +141,76 @@ enum exit_codes {
...
+		if (!rc) {
+			len -= to_read;
+			output += to_read;
+		} else if (rc < 0) {
+			/* Some error happened inside entropy_get_entropy():
+			 * clear the gathered entropy before returning to
+			 * avoid leaking unused information.
+			 */
+			memset(orig_output, 0, orig_len);
+			__asm__ volatile("" : : "g"(orig_output) : "memory");
```
Comment (andrewboie) [inline samples/net/mbedtls_sslclient/src/mini_client.c]: is this portable?
maybe use one of our barrier macros instead?

File: samples/net/mbedtls_sslclient/src/mini_client.c [hunk 2]
```diff
@@ -139,28 +141,76 @@ enum exit_codes {
...
 
-	if (len > sizeof(seed)) {
-		len = sizeof(seed);
+		return 0;
 	}
 
-	memcpy(output, &seed, len);
+	/* If an entropy device has been found during boot, use it to gather
+	 * entropy.  This loop is required since `len` is a `size_t`, and the
+	 * equivalent parameter type in `entropy_get_entropy()` is a `u16_t`.
```
Comment (andrewboie) [inline samples/net/mbedtls_sslclient/src/mini_client.c]: just curious why is the parameter to entropy_get_entropy() a u16_t anyway?

Comment (lpereira) [reply samples/net/mbedtls_sslclient/src/mini_client.c]: That's how the API has been defined.

**Pull request merged**